### PR TITLE
fix(sdk):recursion op support *args

### PR DESCRIPTION
**Pull request by czz2018**
I had problem when I use recursion operator  by variable arguments (*args).

- Optimize the implementation of _graph_component in _component.py mode to support *args.

### CODE REVIEW

File: sdk/python/kfp/dsl/_component.py [hunk 1]
```diff
@@ -143,13 +143,13 @@ def flip_component(flip_result):
...
   @wraps(func)
   def _graph_component(*args, **kargs):
-    # We need to make sure that the arguments are correctly mapped to inputs
-    # regardless of the passing order
-    signature = inspect.signature(func)
-    bound_arguments = signature.bind(*args, **kargs)
     graph_ops_group = Graph(func.__name__)
-    graph_ops_group.inputs = list(bound_arguments.arguments.values())
-    graph_ops_group.arguments = bound_arguments.arguments
+    graph_ops_group.inputs = list(args) + list(kargs.values())
```
Comment (Ark-kun) [inline sdk/python/kfp/dsl/_component.py:147]: I'm not sure this is correct. The ordering of `kargs.values()` is not guaranteed to be stable and more importantly it's not guaranteed to correspond to the input order. This is the reason why we're using `signature.bind`.

**Comment by Ark-kun** [comment]
Thank you for your proposal, but I'm not sure we consider this change an improvement when considering all user scenarios.
* Using positional arguments makes the pipelines less stable, since the argument mapping can silently shift when new inputs are introduced.
* Every component should have a defined interface (inputs and outputs). But this change enables a scenario where the component inputs are not really known until it's instantiated. And it can be instantiated with different arguments sets.
* Recursive graph components are pretty hard to grasp. Having explicit input passing helps making them more readable.

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Comment by chensun** [comment]
> /ok-to-test

@Arhell please do not abuse the review system.

**Comment by rimolive** [comment]
Closing this PR. No activity for more than a year.

/close

**Pull request closed without merging**